### ISSUE TITLE
Port twisted.names.cache to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6093]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6093|
|Type|enhancement|
|Created|2012-10-10 13:47:02Z|
|Branch|https://github.com/twisted/twisted/tree/cache-py3-6093|

This is (sneakily) needed by twisted.names.client (createResolver imports it).

<details><summary>Searchable metadata</summary>

```
trac-id__6093 6093
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_cache_py3_6093 branches/cache-py3-6093
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__names names
keywords__ 
time__1349876822000000 1349876822000000
changetime__1349893596000000 1349893596000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [36052]) Branching to 'cache-py3-6093'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @itamarst|
|-|-|

Quite straight-forward.  Does not have full coverage, but all the parts `twisted.names.client` cares about are covered.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/cache-py3-6093)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @exarkun|
|-|-|

`__future__` imports, and maybe open ticket for missing coverage? Then merge. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [36061]) Merge cache-py3-6093

Author: exarkun
Reviewer: itamarst
Fixes: [#6093](https://github.com/twisted/twisted/issues/6093)

Port `twisted.names.cache` to Python 3.
